plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/internal/EclipseBasedN4JSWorkspace.java

```diff
@@ -83,6 +83,18 @@ public URI findProjectWith(URI nestedLocation) {
 				&& nestedLocation.segmentCount() >= DIRECT_RESOURCE_IN_PROJECT_SEGMENTCOUNT) {
 			return URI.createPlatformResourceURI(nestedLocation.segment(1), true);
 		}
+		// this might happen if the URI was located from non-platform information, e.g. in case
+		// of a source file location found in a soure map
+		if (nestedLocation.isFile()) {
+			String nested = nestedLocation.toString();
+			for (IProject proj : workspace.getProjects()) {
+				URI projURI = URI.createFileURI(proj.getLocation().toFile().toString());
+				String ps = projURI.toString();
+				if (nested.startsWith(ps)) {
+					return projURI;
+				}
+			}
+		}
```

Comment:
This change implies that we will iterate over all workspace projects, whenever `N4JSModel#findProjectWith` is invoked with a `file:`-based URI. According to my understanding, this will happen for all resources in the external library workspace (e.g. validation runs in external library workspace).

Reply:
slightly changed that, also fixing the failing tests @yoosiba mentioned. If location is file, we first assume it is in the external library workspace. If it is not find there, only then we try the "normal" workspace and recreate the URI to become a platform URI. This is not 100% bullet proof, in the long run we have to unify the file handling.